homeassistant/components/spotify/media_player.py

```diff
@@ -474,6 +474,10 @@ async def async_browse_media(self, media_content_type=None, media_content_id=Non
         """Implement the websocket media browsing helper."""
 
         if not self._scope_ok:
+            _LOGGER.warning(
```

Comment:
This is incorrect. The integration should trigger a re-auth when the scope isn't ok.

Reply:
fair enough 👍🏾

Reply:
So I did some digging in the code. Am I correct to think that the re-auth should be initialled via config_flow as described here: https://developers.home-assistant.io/docs/data_entry_flow_index#initializing-a-config-flow-from-an-external-source https://github.com/home-assistant/core/blob/d156e95f8a3e0871ca5658829f1e2a9243c7b519/homeassistant/components/spotify/config_flow.py#L61

Reply:
I'm getting lost in the code. Where is the flow_manager? Sorry fist time contributor.  I Tried the docs but could not find a guide that could help me. Maybe its there but I just did not find it. Could you provide some pointers?

Reply:
also can't seem to find the usage of `async_step_reauth` 🤔 🤯

Reply:
https://github.com/home-assistant/core/blob/bf9f3e1997a56760b9f378a38547149264928436/homeassistant/components/spotify/__init__.py#L80-L87

Reply:
Ok something fishy is going on then. Added more info into the linked issue. I'll try to reproduce the issue on my dev instance and continue debugging the code from there. Thx for the pointer 😊.